### PR TITLE
SS-694 Create user profile page

### PR DESCRIPTION
**Pull request by alfredeen**
## Description

This PR adds a user profile page. The affiliation short code is converted and displayed as friendly name via ajax. This PR also adds two new API endpoints for universities and departments lookups. 

Jira [link](https://scilifelab.atlassian.net/browse/SS-694)

## Checklist

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [x] I have included a link to the issue on GitHub or JIRA (if any)
- [ ] I have included migration files (if there are changes to the model classes)
- [x] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [ ] I have updated the related documentation (if necessary)
- [ ] I have updated the release notes (releasenotes.md)
- [x] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [ ] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

### CODE REVIEW

File: api/openapi/lookups_api.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        Without a query string parameter it returns the entire list.
+
+        :returns list of dict or dict: The dict contains attributes code and name.
+        """
+        print("UniversityLookupAPI. Entered list_or_single")
+
+        try:
+            universities = self.__get_universities()
+        except Exception as e:
+            print(f"Exception: {e}")
```
Comment (churnikov) [inline api/openapi/lookups_api.py:39]: I wish we would at some point be doing proper logging.

But with current setup it's good enough :)

File: studio/views.py [hunk 2]
```diff
@@ -88,3 +93,40 @@ def get(self, request, format=None):
...
+
+
+@login_required
+def profile(request):
+    # Get the user profile
+    try:
+        # Note that not all users have a user profile object
+        # such as the admin superuser
+        user_profile = UserProfile.objects.get(user_id=request.user.id)
+    except:  # noqa E722 OK here
```
Comment (churnikov) [inline studio/views.py:105]: Is it though?
Can't we explicitly catch `NotFound` error or something? At least `Exception`

**Review by churnikov — COMMENTED** [review]
These are some nice, clean api routes, I like it :)
Maybe we could use them for sign up form?

File: api/openapi/lookups_api.py (hunk 1 above)
Comment (alfredeen) [reply api/openapi/lookups_api.py:39]: I completely agree about adding real logging. And auditing.

File: studio/views.py (hunk 2 above)
Comment (alfredeen) [reply studio/views.py:105]: I will add this

**Comment by alfredeen** [comment]
> These are some nice, clean api routes, I like it :) Maybe we could use them for sign up form?

I agree, though I see that the current form uses Django ModelForm so not as straightforward of a swap as maybe we want to do at this time, now that the current approach is tested and we want to release soon.

File: templates/user/profile.html [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                        <div class="row pt-1">
+                            <div class="col-4">Affiliation</div>
+                            <div id="userAffiliation" class="col-8 fw-bold">{{user_profile.affiliation}}</div>
+                        </div>
+                        <div class="row pt-1">
+                            <div class="col-4">Department</div>
+                            <div class="col-8 fw-bold">{{user_profile.department}}</div>
+                        </div>
+
+                        <div class="row mt-5 p-3 small text-white" style="background-color: #82AEB2;">
```
Comment (sandstromviktor) [inline templates/user/profile.html:48]: Let's use the css classes instead.
```suggestion
                        <div class="row mt-5 p-3 small text-white bg-teal-050">
```

**Review by sandstromviktor — CHANGES_REQUESTED** [review]
Good job Johan! I added a specific suggested change, and a comment about using divs in rows.

File: templates/user/profile.html (hunk 3 above)
Comment (alfredeen) [reply templates/user/profile.html:48]: Good suggestions. I will fix this.

File: templates/user/profile.html [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                            <p>Last login: {{user.last_login}} </p>
+                            <p>Account created: {{user.date_joined.date}}</p>
```
[the remark this answers is not in the document]
Comment (alfredeen) [reply templates/user/profile.html]: For the user profile I prefer the larger area in the bottom with user account information and we can in the future add more information of interest.

**Review by sandstromviktor — APPROVED** [review]
Very nice Johan!

**Pull request merged**